### DISCUSSION TITLE
Symlink not being created for my Python install

### DISCUSSION THREAD

**Discussion by SirRomey** [Homebrew/discussions#3696] [Getting started]
Hey,

So I just got a new MacBook and was setting everything up. When I installed Python, it gave me this message.
``Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /opt/homebrew/opt/python@3.10/libexec/bin``

But for some reason these symlinks do not work at all. Is there something I'm doing wrong? If anyone got any suggestions or fixes please drop them here.

**Comment by carlocab** [comment]
What do you mean by "do not work at all"? What commands did you run? What happened when you ran them? What did you expect to happen?

**Reply by SirRomey** [reply]
Oh right.. sorry I didn't make it clear. I've got a couple of pre-build scripts that I ported over. Essentially, they are similar to ``pip install...``, ``python ...``. 

From my understanding the symlink was supposed to make it possible for these scripts to run fine as they are essentially linked their Python3 counterpart. But it does not work. I could just change the scripts but I wanna see if I could get this to work.

**Reply by carlocab** [reply]
Is `/opt/homebrew/opt/python@3.10/libexec/bin` in your `PATH`?

**Reply by SirRomey** [reply]
This is my ``$PATH``,
``/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/opt/homebrew/sbin``

**Reply by carlocab** [reply]
You need to add `/opt/homebrew/opt/python@3.10/libexec/bin` to your `PATH` to be able to call `python3`, `pip3`, etc., as `python` and `pip`, and so on.

**Reply by SirRomey** [reply]
This might be out of context but could you let me know how to add to PATH? I've tried a bunch of stuff I found on StackOverflow but when I do ``echo $PATH`` it does not show up. (just got a MacBook and still getting the hang of it)

**Reply by carlocab** [reply]
Try https://osxdaily.com/2014/08/14/add-new-path-to-path-command-line/

**Reply by SirRomey** [reply]
Alright.. I've added it but it still does not seem to work.
```sirromey@Aadhavs-MacBook-Air ~ % echo $PATH
/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/sirromey/opt/homebrew/opt/python@3.10/libexec/bin
sirromey@Aadhavs-MacBook-Air ~ % pip
zsh: command not found: pip```

**Reply by carlocab** [reply]
This:
```
/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/sirromey/opt/homebrew/opt/python@3.10/libexec/bin
```
looks wrong. It seems like the last entry should be
```
/opt/homebrew/opt/python@3.10/libexec/bin
```
and not
```
/Users/sirromey/opt/homebrew/opt/python@3.10/libexec/bin
```

**Reply by SirRomey** [reply]
Ohh.. my bad. I add the ``~`` and it routed it to the user I think. It works fine now. Thanks a lot. If possible, it would be best if brew could do it automatically while installing.

**Reply by carlocab** [reply]
No, it wouldn't. `brew` should only write inside `HOMEBREW_PREFIX` (in your case, this is `/opt/homebrew`).

Doing this automatically is a recipe for bugs and security vulnerabilities.
